core/src/test/java/apoc/trigger/TriggerNewProceduresTest.java

```diff
@@ -532,4 +567,59 @@ public void testInstallTriggerInSystemDb() {
         }
     }
 
+    @Test
+    public void testEventualConsistency() {
+        LongStream.range(0, 100).forEach(i -> {
+            final String name = UUID.randomUUID().toString();
+            final String query = "UNWIND $createdNodes AS n SET n.count = " + i;
+    
+            // this does nothing, just to test consistency
+            sysDb.executeTransactionally("CALL apoc.trigger.install('neo4j', $name, 'return 1', {phase: 'afterAsync'})",
+                    map("name", UUID.randomUUID().toString()) );
+
+            // create trigger
+            sysDb.executeTransactionally("CALL apoc.trigger.install('neo4j', $name, $query, {phase: 'afterAsync'})",
+                    map("name", name, "query", query));
+            awaitTriggerDiscovered(db, name, query);
+            db.executeTransactionally("CREATE (n:Something)");
+
+            // check trigger
+            assertEventually(() ->
+                            db.executeTransactionally("MATCH (c:Something) RETURN c.count as count", Map.of(),
+                                    result -> {
+                                        final ResourceIterator<Object> count = result.columnAs("count");
+                                        return Objects.equals(count.next(), i);
+                                    }), 
+                    (v) -> v, 30L, TimeUnit.SECONDS);
+
+            testCall(db, "MATCH (c:Something) RETURN c.count as count",
+                    (row) -> assertEquals(i, row.get("count")));
+
+            // stop trigger
+            sysDb.executeTransactionally("CALL apoc.trigger.stop('neo4j', $name)",
+                    map("name", name, "query", query));
+            awaitTriggerDiscovered(db, name, query, true);
+
+            // this does nothing, just to test consistency
+            sysDb.executeTransactionally("CALL apoc.trigger.install('neo4j', $name, 'return 1', {phase: 'afterAsync'})",
+                    map("name", UUID.randomUUID().toString()) );
+
+            // start trigger
+            sysDb.executeTransactionally("CALL apoc.trigger.start('neo4j', $name)",
+                    map("name", name, "query", query));
+            awaitTriggerDiscovered(db, name, query);
+            
+            // this does nothing, just to test consistency
+            sysDb.executeTransactionally("CALL apoc.trigger.install('neo4j', $name, 'return 1', {phase: 'afterAsync'})",
+                    map("name", UUID.randomUUID().toString()) );
+
+            // drop trigger and check if dropped
+            sysDb.executeTransactionally("CALL apoc.trigger.drop('neo4j', $name)", Map.of("name", name));
+            testCallCountEventually(db, "CALL apoc.trigger.list() yield name where name = $name return name",
+                    Map.of("name", name),
+                    0, TIMEOUT);
```

Comment:
This probably ought to be the new show procedure since I don't think we ought to use a mix of new and deprecated procedures in the same test. What do you think?

Reply:
I think it depends on [this](https://github.com/neo4j-contrib/neo4j-apoc-procedures/pull/3335#pullrequestreview-1194406053). 
I mean, the `apoc.trigger.show` currently check if the trigger is picked up, but it may be that it has not actually been updated in the event listener cache (i.e. in `activeTriggers` map, present in `TriggerHandler.java`). 
So the `apoc.trigger.list` in this case is more accurate for these tests, because it verifies that the event listener really has the updated trigger. 
For this reason, the apoc.trigger.list is not deprecated.